src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -691,13 +650,51 @@ protected internal bool IsArc()
 
         #region Private Methods
 
-        protected internal bool SetExecutePermissionForProxyOnLinux(string path)
+        private string SearchForInstalledProxyPath()
+        {
+            var results = InvokeCommand.InvokeScript(
+                script: "Get-module -ListAvailable -Name Az.Ssh.ArcProxy");
+
+            foreach (var result in results)
+            {
+                if (result?.BaseObject is PSModuleInfo moduleInfo)
+                {
+                    var proxyPath = GetProxyPathInModuleDirectory(moduleInfo.Path);
+
+                    if (moduleInfo.Version >= new Version("2.0.0"))
+                    {
+                        WriteWarning(String.Format(Resources.UnsuportedVersionProxyModule, moduleInfo.Path, moduleInfo.Version));
+                        continue;
+                    }
+
+                    if (!File.Exists(proxyPath))
+                    {
+                        continue;
+                    }
+
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                    {
+                        // does this check even work?
```

Comment:
Do you mean `SetExecutePermissionForProxyOnLinux()` always returns true?

Reply:
I wasn't able to confirm until a bit ago, but that seems to be the case. The InvokeScript method is not able to determine if the chmod operation failed. It still prints the error message from the chmod command, but it won't throw an exception.

Reply:
It may be worth checking the `$LASTEXITCODE` variable in the script and manually throw an exception if it's non-zero. For the time being I'm OK to approve and merge since that's not what's introdcued by this PR.